src/main/java/org/jabref/gui/actions/ActionHelper.java

```diff
@@ -43,6 +46,28 @@ public static BooleanExpression needsStudyDatabase(StateManager stateManager) {
         return BooleanExpression.booleanExpression(binding);
     }
 
+    // Makes sure there is at least one group selected, and if there are multiple groups selected
+    // all have the same parent node.
+    public static BooleanExpression needsSelectedGroupsShareParent(StateManager stateManager) {
+        if (stateManager.activeGroupProperty().isEmpty()) {
```

Comment:
This reads strange. A `BooleanExpression` is someething evaluated "live" during the run of the program. With the currente change, this live property is different depending on a condition on its creation.

Maybe, the whole `if` can be done. The `return` thing reads OK.

Reply:
That was an (admittedly short-sighted) attempt to get around the fact that the `.and()` function does not short-circuiting, so even if it failed the check for `size(activeGroupProperty()).greaterThan(1)` it would try to get the first element of the list and raise a null-pointer exception. As best I can tell with my debugger anyways, both the context menus in the Groups side view and the Main Table are re-created each time a right-click is registered, so I went ahead and got around the issue with the non-dynamic `if`. Looking closer, it does seem that the original `Add Groups` button at the bottom is the only one that is static in it's implementation (though since it's set to always add to root, that technically wouldn't be an issue).